Code/Engine/Foundation/SimdMath/Implementation/AVX/AVXDouble_inl.h

```diff
@@ -0,0 +1,318 @@
+{
+  m_v = _mm256_div_pd(m_v, d.m_v);
+  return *this;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::IsEqual(const ezSimdDouble& rhs, const ezSimdDouble& fEpsilon) const
+{
+  ezSimdDouble minusEps = rhs - fEpsilon;
+  ezSimdDouble plusEps = rhs + fEpsilon;
+  return ((*this >= minusEps) && (*this <= plusEps));
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator==(const ezSimdDouble& d) const
+{
+  return _mm_comieq_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator!=(const ezSimdDouble& d) const
+{
+  return _mm_comineq_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>=(const ezSimdDouble& d) const
+{
+  return _mm_comige_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>(const ezSimdDouble& d) const
+{
+  return _mm_comigt_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<=(const ezSimdDouble& d) const
+{
+  return _mm_comile_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<(const ezSimdDouble& d) const
+{
+  return _mm_comilt_sd(_mm256_castpd256_pd128(m_v), _mm256_castpd256_pd128(d.m_v)) == 1;
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator==(double d) const
+{
+  return (*this) == ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator!=(double d) const
+{
+  return (*this) != ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>(double d) const
+{
+  return (*this) > ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>=(double d) const
+{
+  return (*this) >= ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<(double d) const
+{
+  return (*this) < ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<=(double d) const
+{
+  return (*this) <= ezSimdDouble(d);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator==(float f) const
+{
+  return (*this) == ezSimdDouble(f);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator!=(float f) const
+{
+  return (*this) != ezSimdDouble(f);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>(float f) const
+{
+  return (*this) > ezSimdDouble(f);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator>=(float f) const
+{
+  return (*this) >= ezSimdDouble(f);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<(float f) const
+{
+  return (*this) < ezSimdDouble(f);
+}
+
+EZ_ALWAYS_INLINE bool ezSimdDouble::operator<=(float f) const
+{
+  return (*this) <= ezSimdDouble(f);
+}
+
+template <>
+EZ_ALWAYS_INLINE ezSimdDouble ezSimdDouble::GetReciprocal<ezMathAcc::FULL>() const
+{
+  return _mm256_div_pd(_mm256_set1_pd(1.0), m_v);
+}
+
+template <>
+EZ_ALWAYS_INLINE ezSimdDouble ezSimdDouble::GetReciprocal<ezMathAcc::BITS_23>() const
```

Comment:
We can remove this precision template for this function (sqrt and invsqrt as well) in the double version and always use full precision. It doesn't make much sense for double and will most likely not be any faster than the full precision.